lib/elixir/src/elixir_expand.erl

```diff
@@ -69,7 +69,17 @@ expand({alias, Meta, [Ref, Opts]}, E) ->
 
   if
     is_atom(ERef) ->
-      {ERef, expand_alias(Meta, true, ERef, EOpts, ET)};
+      case lists:keyfind(as, 1, Opts) of
+        {as, _} ->
+          {ERef, expand_alias(Meta, true, ERef, EOpts, ET)};
+        _->
+          case atom_to_list(ERef) of
+            "Elixir." ++ ([FirstLetter | _]) when FirstLetter >= $A, FirstLetter =< $Z ->
+              {ERef, expand_alias(Meta, true, ERef, EOpts, ET)};
+            _ ->
+              form_error(Meta, E, ?MODULE, {invalid_alias_module, Ref})
+          end
```

Comment:
This looks good but you likely want to do it on elixir_aliases:last, by changing it to return `{:ok, _}` or `:error`, and then you change the callers of `elixir_aliases:last` accordingly.

Reply:
Sure, will refactor it to that, thanks for the heads up!